sdk/identity/Azure.Identity/src/Credentials/SharedTokenCacheCredentialOptions.cs

```diff
@@ -44,9 +47,18 @@ public string TenantId
         public AuthenticationRecord AuthenticationRecord { get; set; }
 
         /// <summary>
-        /// Specifies the <see cref="TokenCachePersistenceOptions"/> to be used by the credential.
+        /// Specifies the <see cref="TokenCachePersistenceOptions"/> to be used by the credential. Value cannot be null.
         /// </summary>
-        public TokenCachePersistenceOptions TokenCachePersistenceOptions { get; }
+        public TokenCachePersistenceOptions TokenCachePersistenceOptions
+        {
+            get { return _tokenCachePersistenceOptions; }
+            set
+            {
+                Argument.AssertNotNull(value, nameof(value));
```

Comment:
This property used to only be settable through a constructor overload which ensures it always has a value. There are other places in the code which assume this will always have a value. I figured the easiest way to ensure adding a setter didn't regress anything was to prevent the value from being set to null. If you disagree I can reevaluate the other changes which would be needed to account for this possibly being null.